From 49ae2888b43cad358e2ff60a69722341116e7b40 Mon Sep 17 00:00:00 2001
From: Xin LI <d@delphij.net>
Date: Sun, 02 Oct 2011 15:02:18 +0000
Subject: 	Avoid problems when FreeBSD goes to version 10.

	* libltdl/m4/libtool.m4: Remove mentions of freebsd1. Use dots
	for FreeBSD 2 and 3 detection to avoid future problems
	* libltdl/m4/ltmain.m4sh: ditto.
---
diff --git a/libltdl/config/ltmain.m4sh b/libltdl/config/ltmain.m4sh
index 511480f..be37978 100644
--- libltdl/config/ltmain.m4sh
+++ libltdl/config/ltmain.m4sh
@@ -2686,7 +2686,7 @@ static const void *lt_preloaded_setup() {
 	  # linked before any other PIC object.  But we must not use
 	  # pic_flag when linking with -static.  The problem exists in
 	  # FreeBSD 2.2.6 and is fixed in FreeBSD 3.1.
-	  *-*-freebsd2*|*-*-freebsd3.0*|*-*-freebsdelf3.0*)
+	  *-*-freebsd2.*|*-*-freebsd3.0*|*-*-freebsdelf3.0*)
 	    pic_flag_for_symtable=" $pic_flag -DFREEBSD_WORKAROUND" ;;
 	  *-*-hpux*)
 	    pic_flag_for_symtable=" $pic_flag"  ;;
diff --git a/libltdl/m4/libtool.m4 b/libltdl/m4/libtool.m4
index 350fbe1..0de7f3c 100644
--- libltdl/m4/libtool.m4
+++ libltdl/m4/libtool.m4
@@ -2475,7 +2475,7 @@ freebsd* | dragonfly*)
     objformat=`/usr/bin/objformat`
   else
     case $host_os in
-    freebsd[[123]]*) objformat=aout ;;
+    freebsd[[23]].*) objformat=aout ;;
     *) objformat=elf ;;
     esac
   fi
@@ -2493,7 +2493,7 @@ freebsd* | dragonfly*)
   esac
   shlibpath_var=LD_LIBRARY_PATH
   case $host_os in
-  freebsd2*)
+  freebsd2.*)
     shlibpath_overrides_runpath=yes
     ;;
   freebsd3.[[01]]* | freebsdelf3.[[01]]*)
@@ -5200,7 +5200,7 @@ _LT_EOF
       ;;
 
     # Unfortunately, older versions of FreeBSD 2 do not have this feature.
-    freebsd2*)
+    freebsd2.*)
       _LT_TAGVAR(archive_cmds, $1)='$LD -Bshareable -o $lib $libobjs $deplibs $linker_flags'
       _LT_TAGVAR(hardcode_direct, $1)=yes
       _LT_TAGVAR(hardcode_minus_L, $1)=yes
@@ -6206,7 +6206,7 @@ if test "$_lt_caught_CXX_error" != yes; then
         esac
         ;;
 
-      freebsd[[12]]*)
+      freebsd2.*)
         # C++ shared libraries reported to be fairly broken before
 	# switch to ELF
         _LT_TAGVAR(ld_shlibs, $1)=no
--
cgit v0.8.3.4
